talk/morelanguage/raii.tex

```diff
@@ -189,9 +190,34 @@
       auto a = std::make_unique<Foo>(arg1, arg2);
       std::cout << a.get() << " points to "
                 << a->someMember << '\n';
-      // allocation of an array of Foos
-      // calls default constructor
+
+      // deallocations at end of scope
+    \end{cppcode*}
+    \begin{minted}[autogobble=true]{text}
+
+      0x6000005fe9a0 points to 5
+    \end{minted}
+  \end{exampleblock}
+\end{frame}
+
+\begin{frame}[fragile]
+  \frametitlecpp[11]{Arrays and unique\_ptr}
+  \begin{block}{Dynamic arrays}
+    \begin{itemize}
+      \item \mintinline{cpp}{unique_ptr} can wrap arrays
+      \item Use \mintinline{cpp}{T[]} as template parameter
+      \item This will enable the index access operator
+      \item The default constructor is called for each element
+      \item If size known at compile time $\rightarrow$ \mintinline{cpp}{std::array}
+      \item If size might change $\rightarrow$ \mintinline{cpp}{std::vector}
```

Comment:
I guess those are recommendations? Maybe even best practices :) Could you squeeze a "prefer" here so we know it's a recommendation?
```suggestion
      \item If size known at compile time prefer $\rightarrow$ \mintinline{cpp}{std::array}
      \item If size might change prefer $\rightarrow$ \mintinline{cpp}{std::vector}
```